### PR TITLE
Add UI test to NavigationRail

### PR DESCRIPTION
**Pull request by khalp**
Adds UI tests to NavigationRail and makes some small fixes to project structure and a small bug introduced in the DualView UI test PR. A few new functions were also added to TestUtils for getting string resources with arguments and simulating an "unfold"

### CODE REVIEW

File: NavigationRail/src/androidTest/java/com/microsoft/device/display/samples/navigationrail/DetailTest.kt [hunk 1]
```diff
@@ -0,0 +1,226 @@
...
+
+        // Assert that back button does not exist
+        composeTestRule.onNodeWithContentDescription(composeTestRule.getString(R.string.back)).assertDoesNotExist()
+    }
+
+    /**
+     * Tests that the back button successfully returns to the gallery view from the item detail view
+     */
+    @Test
+    fun app_backButtonChangesPanes() {
```
Comment (joyl1216) [inline NavigationRail/src/androidTest/java/com/microsoft/device/display/samples/navigationrail/DetailTest.kt:95]: Can we rename this by adding "InSingleScreenMode" and the comment as well?

File: NavigationRail/src/androidTest/java/com/microsoft/device/display/samples/navigationrail/DetailTest.kt [hunk 2]
```diff
@@ -0,0 +1,226 @@
...
+        // Assert drawer is collapsed again
+        composeTestRule.onNodeWithTag(composeTestRule.getString(R.string.content_drawer))
+            .assertDrawerStateEquals(DrawerState.Collapsed)
+    }
+
+    /**
+     * Tests that the info box fact contents are horizontally scrollable
+     */
+    @Test
+    fun detailView_factsAreHorizontallyScrollable() {
```
Comment (joyl1216) [inline NavigationRail/src/androidTest/java/com/microsoft/device/display/samples/navigationrail/DetailTest.kt:160]: I know it is not testing-related, but I couldn't remember why the info box has to be horizontally scrollable.

File: NavigationRail/src/androidTest/java/com/microsoft/device/display/samples/navigationrail/NavComponentTest.kt [hunk 3]
```diff
@@ -0,0 +1,77 @@
...
+ * Licensed under the MIT License.
+ */
+
+package com.microsoft.device.display.samples.navigationrail
+
+import androidx.compose.animation.ExperimentalAnimationApi
+import androidx.compose.foundation.ExperimentalFoundationApi
+import androidx.compose.material.ExperimentalMaterialApi
+import androidx.compose.runtime.getValue
+import androidx.compose.runtime.setValue
```
Comment (joyl1216) [inline NavigationRail/src/androidTest/java/com/microsoft/device/display/samples/navigationrail/NavComponentTest.kt:12]: I don't know why `ktlint` check didn't pick them up, but we don't need to import `getValue` and `setValue` here.

File: NavigationRail/src/androidTest/java/com/microsoft/device/display/samples/navigationrail/DetailTest.kt (hunk 1 above)
Comment (khalp) [reply NavigationRail/src/androidTest/java/com/microsoft/device/display/samples/navigationrail/DetailTest.kt:95]: Not too long at all, good catch!

File: NavigationRail/src/androidTest/java/com/microsoft/device/display/samples/navigationrail/DetailTest.kt (hunk 2 above)
Comment (khalp) [reply NavigationRail/src/androidTest/java/com/microsoft/device/display/samples/navigationrail/DetailTest.kt:160]: This is mainly for the 6.7 emulator, where sometimes the second fact is cut off:
image

**Comment by joyl1216** [comment]
The PR looks good. I would suggest that we add some test cases to test the navigationRail and bottomNavigation on dual screen and single screen, since it is the main scenario for this sample. And also test whether the list is showing across both screens, not detail view shown in dual landscape mode(horizontal fold).

**Review by joyl1216 — APPROVED** [review]

**Pull request merged**